sync-android/src/test/java/com/cloudant/sync/datastore/encryption/CacheKeyProviderTests.java

```diff
@@ -0,0 +1,121 @@
+package com.cloudant.sync.datastore.encryption;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeNotNull;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+public class CacheKeyProviderTests {
+    @Before
+    public void beforeMethod() {
+        assumeNotNull(ProviderTestUtil.getContext());
+    }
+
+    @Test
+    public void testCreateProviderWithIdentifier() {
+        CacheKeyProvider provider = new CacheKeyProvider(new AndroidKeyProvider(ProviderTestUtil
+                .getContext(),
+                ProviderTestUtil.password, ProviderTestUtil.getUniqueIdentifier()));
+
+        // This will cause the provider to generate a key using the AndroidKeyProvider. The key
+        // will be encrypted, persisted, and stored in memory.
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        // This will cause the provider to load the cached key.
+        EncryptionKey cachedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not cached", cachedKey);
+
+        // Assert cached key and created key are the same
+        assertTrue("cachedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                cachedKey.getKey()));
+
+        AndroidKeyProvider androidKeyProvider = (AndroidKeyProvider) provider.getKeyProvider();
+        androidKeyProvider.getManager().clearKey();
+    }
+
+    @Test
+    public void testCreateTwoProvidersWithSameIdentifier() {
+        String identifier = ProviderTestUtil.getUniqueIdentifier();
+        CacheKeyProvider provider1 = new CacheKeyProvider(new AndroidKeyProvider(ProviderTestUtil
+                .getContext(),
+                ProviderTestUtil.password, identifier));
+        CacheKeyProvider provider2 = new CacheKeyProvider(new AndroidKeyProvider(ProviderTestUtil
+                .getContext(),
+                ProviderTestUtil.password, identifier));
+
+        EncryptionKey provider1Key = provider1.getEncryptionKey();
+        assertNotNull("provider1 EncryptionKey was not created", provider1Key);
+
+        EncryptionKey provider2Key = provider2.getEncryptionKey();
+        assertNotNull("provider2 EncryptionKey was not created", provider2Key);
+
+        assertTrue("provider1 key and provider2 key should be equal", Arrays.equals(provider1Key
```

Comment:
Similar to above, you can assert that the pointers `provider1Key` and `provider2Key` are different here